WordPress/wordpress-develop
# Various CS/QA fixes (Trac: 50767)

### CS: use `self` when appropriate

* `get_default_primary_column_name()` is a protected method, so calling it statically with the class name is bad practice.
* Similarly this applies when calling a `private` constructor.

### QA: don't use alias functions

Using the canonical function name for PHP functions is strongly recommended as aliases may be deprecated or removed without (much) warning.

This replaces all uses of the following:
* `join()` with `implode()`
* `sizeof()` with `count()`
* `is_writeable()` with `is_writable()`
* `doubleval()` with a `(float)` cast

In part, this is a follow-up to 47746

### QA: remove assumption in code

Use a more stable condition.
If `$wpdb->get_row()` is successful and the `$output` parameter has not been set, the output will be an instance of `stdClass`, so test to confirm that instead of testing against "not null".

### Modernize: use `instanceof` instead of a comparison with `get_class()` 

Includes adjusting external libraries which are no longer maintained externally.

Trac ticket: https://core.trac.wordpress.org/ticket/50767

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**
